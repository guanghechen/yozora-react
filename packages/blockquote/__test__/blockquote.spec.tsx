import { mount, render } from 'enzyme'
import React from 'react'
import Blockquote from '../src'

describe('prop types', function () {
  beforeEach(() => {
    jest.spyOn(global.console, 'error').mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })
  })

  it('forward ref', function () {
    const ref = React.createRef<HTMLHRElement>()
    const wrapper = mount(
      <Blockquote ref={ref} data-name="yozora-blockquote">
        Blockquote contents.
      </Blockquote>,
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-name')).toEqual('yozora-blockquote')
  })

  it('children is required', function () {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        render(<Blockquote>{value}</Blockquote>)
      }).toThrow(/The prop `children` is marked as required/i)
    }

    expect(render(<Blockquote>Hello, world!</Blockquote>).text()).toEqual(
      'Hello, world!',
    )
  })

  it('className is optional', function () {
    expect(
      render(<Blockquote>Blockquote contents.</Blockquote>).hasClass(
        'yozora-blockquote',
      ),
    ).toEqual(true)

    expect(
      render(
        <Blockquote className="my-blockquote">Blockquote contents.</Blockquote>,
      ).hasClass('my-blockquote'),
    ).toEqual(true)
  })
})

describe('snapshot', function () {
  it('default', function () {
    const wrapper = render(
      <Blockquote>
        some text1
        <span>some text2</span>
      </Blockquote>,
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('custom', function () {
    const wrapper = render(
      <Blockquote
        className="custom-class"
        data-name="yozora-blockquote"
        style={{ color: 'orange' }}
      >
        some text1
        <span>some text2</span>
        <Blockquote>some text3</Blockquote>
      </Blockquote>,
    )
    expect(wrapper).toMatchSnapshot()
  })
})
