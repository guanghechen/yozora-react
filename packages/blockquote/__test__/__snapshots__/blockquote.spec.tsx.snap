// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic rendering case snapshot 1`] = `
.c0 {
  padding: 0.625em 1em;
  border-left: 0.25em solid transparent;
  margin: 0 0 1.25em;
  background: none;
  color: inherit;
}

<blockquote
  class="c0"
  style="color:orange;font-size:16px"
>
  some text1
  <span>
    some text2
  </span>
</blockquote>
`;

exports[`basic rendering case snapshot with theme 1`] = `
.c0 {
  padding: 0 1rem;
  border-left: 0.25em solid transparent;
  margin: 18;
  background: rgba(0,0,0,0.15);
  color: red;
}

<blockquote
  class="c0"
>
  some text1
  <span>
    some text2
  </span>
</blockquote>
`;
